website/scripts/send-data-to-vanta.js

```diff
@@ -0,0 +1,288 @@
+module.exports = {
+
+
+  friendlyName: 'Send data to vanta',
+
+
+  description: 'An hourly script that gathers and formats host and user data from Fleet instances sends it to Vanta on behalf of a Vanta user.',
+
+
+  fn: async function () {
+
+    // Find all VantaConnection records with isConnectedToVanta: true
+    let allActiveVantaConnections = await VantaConnection.find({isConnectedToVanta: true});
+
+    sails.log('Syncing Fleet instance data with Vanta for '+allActiveVantaConnections.length+(allActiveVantaConnections.length > 1 ? ' connections.' : ' connection.'));
+
+    // Create an empty object to store caught errors. We don't want this script to stop running if there is an error with a single Vanta integration, so instead, we'll store any errors that occur and stop the script for that connection if any occur, and we'll log them individually before the script is done.
+    let errorReportById = {};
+
+    // use sails.helpers.flow.simutaniouslyForEach to send data for each connection record.
+    await sails.helpers.flow.simultaneouslyForEach(allActiveVantaConnections, async (vantaConnection)=>{
+      let connectionIdAsString = String(vantaConnection.id);
+
+      //  ┬─┐┌─┐┌─┐┬─┐┌─┐┌─┐┬ ┬  ╦  ╦┌─┐┌┐┌┌┬┐┌─┐  ┌┬┐┌─┐┬┌─┌─┐┌┐┌
+      //  ├┬┘├┤ ├┤ ├┬┘├┤ └─┐├─┤  ╚╗╔╝├─┤│││ │ ├─┤   │ │ │├┴┐├┤ │││
+      //  ┴└─└─┘└  ┴└─└─┘└─┘┴ ┴   ╚╝ ┴ ┴┘└┘ ┴ ┴ ┴   ┴ └─┘┴ ┴└─┘┘└┘
+      let authorizationTokenRefreshResponse = await sails.helpers.http.post.with({
+        url:'https://api.vanta.com/oauth/token',
+        data: {
+          client_id: sails.config.custom.vantaAuthorizationClientId, // eslint-disable-line camelcase
+          client_secret: sails.config.custom.vantaAuthorizationClientSecret,// eslint-disable-line camelcase
+          refresh_token: vantaConnection.vantaRefreshToken,// eslint-disable-line camelcase
+          grant_type: 'refresh_token',// eslint-disable-line camelcase
+        },
+        headers: { accept: 'application/json' }
+      }).catch((err)=>{
+        // If an error occurs while sending a request to Vanta, we'll add the error to the errorReportById object, with this connections ID set as the key.
+        errorReportById[connectionIdAsString] = new Error(`Could not refresh the token for Vanta connection (id: ${connectionIdAsString}). Full error: ${err}`);
+      });
+
+      if(errorReportById[connectionIdAsString]){// If there was an error with the previous request, stop the script for this Vanta connection.
```

Comment:
instead of stop the script, I'd say "bail early for this" or "avast for this"   (stop the script sounds like you're killing the whole script)